### PR TITLE
`azurerm_monitor_activity_log_alert` - support for `service_health`

### PR DESCRIPTION
**Pull request by HappyTobi**
Hi all,

just a small pr to add more fine grained settings for "ServiceHealth" alerts.

I changed:
- Add new Rest API for Log Alerts
- Add Documentation for fine grained settings
- Add new ServiceHealth Settings
- Add Tests

New TF style to setup `ServiceHealth`
```
resource "azurerm_monitor_activity_log_alert" "main" {
  name                = "example-activitylogalert"
  resource_group_name = azurerm_resource_group.main.name
  scopes              = [azurerm_resource_group.main.id]
  description         = "This alert will monitor a specific storage account updates."

  criteria {
    category = "ServiceHealth"    
    servicehealth {
      events   = ["Incident", "Maintenance"]
      regions  = ["Global", "West Europe"]
      services = ["App Service (Linux)", "Network Infrastructure"]
    }
  }

  action {
    action_group_id = azurerm_monitor_action_group.main.id
  }
}
```

Added requested feature from issues:
#2996, #7392

Regards,
Tobi

### CODE REVIEW

File: azurerm/internal/services/monitor/monitor_activity_log_alert_resource.go
Comment (ArcturusZhang) [inline azurerm/internal/services/monitor/monitor_activity_log_alert_resource.go:455]: Could we please extract this to an expand function?

File: azurerm/internal/services/monitor/monitor_activity_log_alert_resource.go
Comment (ArcturusZhang) [inline azurerm/internal/services/monitor/monitor_activity_log_alert_resource.go:472]: Could we please extract this to an expand function?

File: azurerm/internal/services/monitor/monitor_activity_log_alert_resource.go
Comment (ArcturusZhang) [inline azurerm/internal/services/monitor/monitor_activity_log_alert_resource.go:447]: Actually we can omit this checking because this has been ensured by the schema

File: azurerm/internal/services/monitor/monitor_activity_log_alert_resource.go
Comment (ArcturusZhang) [inline azurerm/internal/services/monitor/monitor_activity_log_alert_resource.go:458]: Similar here, we could omit this because this has been ensured by the schema

File: azurerm/internal/services/monitor/monitor_activity_log_alert_resource.go
Comment (ArcturusZhang) [inline azurerm/internal/services/monitor/monitor_activity_log_alert_resource.go:154]: Could we change this to `service_health`?
```suggestion
						"service_health": {
```

File: azurerm/internal/services/monitor/monitor_activity_log_alert_resource.go
Comment (ArcturusZhang) [inline azurerm/internal/services/monitor/monitor_activity_log_alert_resource.go:484]: I just notice that we are adding the 3 sets to the same list, then could we just make the expansion of `service_health` in a big expand function?

File: azurerm/internal/services/monitor/monitor_activity_log_alert_resource_test.go
Comment (ArcturusZhang) [inline azurerm/internal/services/monitor/monitor_activity_log_alert_resource_test.go:179]: Actually in the test checkings, for the non-computed attributes, we do not need to check them one by one manually. The importStep will check all of them for you

File: azurerm/internal/services/monitor/monitor_activity_log_alert_resource_test.go
Comment (ArcturusZhang) [inline azurerm/internal/services/monitor/monitor_activity_log_alert_resource_test.go:181]: could we add a `ImportStep` here?

**Review by ArcturusZhang — COMMENTED** [review]
Hey @HappyTobi thanks for this PR!
I just left some minor comments, could you please take a look? Thanks

File: azurerm/internal/services/monitor/monitor_activity_log_alert_resource_test.go
Comment (HappyTobi) [reply azurerm/internal/services/monitor/monitor_activity_log_alert_resource_test.go:181]: Done, thx

File: azurerm/internal/services/monitor/monitor_activity_log_alert_resource_test.go
Comment (HappyTobi) [reply azurerm/internal/services/monitor/monitor_activity_log_alert_resource_test.go:179]: ok, thx I will clean the checks.

File: azurerm/internal/services/monitor/monitor_activity_log_alert_resource.go
Comment (HappyTobi) [reply azurerm/internal/services/monitor/monitor_activity_log_alert_resource.go:154]: Yes fine for me.

File: azurerm/internal/services/monitor/monitor_activity_log_alert_resource.go
Comment (HappyTobi) [reply azurerm/internal/services/monitor/monitor_activity_log_alert_resource.go:458]: thx

File: azurerm/internal/services/monitor/monitor_activity_log_alert_resource.go
Comment (HappyTobi) [reply azurerm/internal/services/monitor/monitor_activity_log_alert_resource.go:447]: thx

File: azurerm/internal/services/monitor/monitor_activity_log_alert_resource.go
Comment (HappyTobi) [reply azurerm/internal/services/monitor/monitor_activity_log_alert_resource.go:472]: Yes, sure

File: azurerm/internal/services/monitor/monitor_activity_log_alert_resource.go
Comment (HappyTobi) [reply azurerm/internal/services/monitor/monitor_activity_log_alert_resource.go:484]: See above

File: azurerm/internal/services/monitor/monitor_activity_log_alert_resource.go
Comment (HappyTobi) [reply azurerm/internal/services/monitor/monitor_activity_log_alert_resource.go:455]: Done thx

**Comment by HappyTobi** [comment]
Hi @ArcturusZhang, @tombuildsstuff 

I updated the code and the documentation. 

Think it's fine now.

Regards,

File: website/docs/r/monitor_activity_log_alert.html.markdown
Comment (jdheyburn) [inline website/docs/r/monitor_activity_log_alert.html.markdown:99]: This is supposed to be `service_health`.

**Review by jdheyburn — CHANGES_REQUESTED** [review]

File: website/docs/r/monitor_activity_log_alert.html.markdown
Comment (HappyTobi) [reply website/docs/r/monitor_activity_log_alert.html.markdown:99]: I will update the docs

Comment (skyflash711) [reply website/docs/r/monitor_activity_log_alert.html.markdown:99]: Are these changes deployed ? if it is yes then how can I use this fine granular piece of code ? Because right now it is giving error "Blocks of type "servicehealth" are not expected here."

Comment (HappyTobi) [reply website/docs/r/monitor_activity_log_alert.html.markdown:99]: @skyflash711 no it's an PR er have to wait till it's reviewed and merged.

After that, a comment from the bot will be added with the version where feature is part of.

**Comment by ClaudiaBaur** [comment]
Is there anything open why this PR cannot be merged? Just asking as we are waiting to consume it... Thanks!

File: azurerm/internal/services/monitor/monitor_activity_log_alert_resource_test.go
Comment (mbfrahry) [reply azurerm/internal/services/monitor/monitor_activity_log_alert_resource_test.go:163]: It looks like this test is failing after the `servicehealth` -> `service_health` change. We'll need that fixed before merging
```
=== CONT  TestAccMonitorActivityLogAlert_basicServiceHealth
    testing.go:620: Step 1/2 error: After applying this test step, the plan was not empty.
        stdout:


        An execution plan has been generated and is shown below.
        Resource actions are indicated with the following symbols:
          ~ update in-place

        Terraform will perform the following actions:

          # azurerm_monitor_activity_log_alert.test will be updated in-place
          ~ resource "azurerm_monitor_activity_log_alert" "test" {
                id                  = "/subscriptions/1a6092a6-137e-4025-9a7c-ef77f76f2c02/resourceGroups/acctestRG-210413143626004304/providers/microsoft.insights/activityLogAlerts/acctestActivityLogAlert-210413143626004304"
                name                = "acctestActivityLogAlert-210413143626004304"
                # (4 unchanged attributes hidden)

              + action {
                  + action_group_id    = "/subscriptions/1a6092a6-137e-4025-9a7c-ef77f76f2c02/resourceGroups/acctestRG-210413143626004304/providers/microsoft.insights/actionGroups/acctestActionGroup1-210413143626004304"
                  + webhook_properties = {}
                }
              - action {
                  - action_group_id = "/subscriptions/1a6092a6-137e-4025-9a7c-ef77f76f2c02/resourceGroups/acctestRG-210413143626004304/providers/microsoft.insights/actionGroups/acctestActionGroup1-210413143626004304" -> null
                }

              ~ criteria {
                    # (1 unchanged attribute hidden)

                  + service_health {
                      + events   = [
                          + "Incident",
                          + "Maintenance",
                        ]
                      + regions  = [
                          + "Global",
                          + "West Europe",
                        ]
                      + services = [
                          + "Action Groups",
                          + "Activity Logs & Alerts",
                        ]
                    }
                }
                # (1 unchanged block hidden)
            }

        Plan: 0 to add, 1 to change, 0 to destroy.
--- FAIL: TestAccMonitorActivityLogAlert_basicServiceHealth (135.66s)
``

**Review by mbfrahry — CHANGES_REQUESTED** [review]
Hey @HappyTobi! Thanks for opening up this PR. It's on the right track but there are a few things I've highlighted below that will need resolved before we can merge

File: azurerm/internal/services/monitor/monitor_activity_log_alert_resource_test.go
Comment (HappyTobi) [reply azurerm/internal/services/monitor/monitor_activity_log_alert_resource_test.go:373]: Thx! I updated it, also the regions -> locations change

File: azurerm/internal/services/monitor/monitor_activity_log_alert_resource_test.go
Comment (HappyTobi) [reply azurerm/internal/services/monitor/monitor_activity_log_alert_resource_test.go:163]: Yes but that was only one thing the current code (released one) also had an issue.
All problems are fixed now.
More tests also added.

**Review by mbfrahry — APPROVED** [review]
LGTM! Thanks for this @HappyTobi

**Pull request merged**
